app/subscriber/src/components/tool-bar/AddToReportMenu.tsx

```diff
@@ -33,82 +47,74 @@ export const AddToReportMenu: React.FC<IAddToReportMenuProps> = ({ content, onCl
-      if (!activeReport) return;
-
+    async (report: IReportModel, sectionName: string, content: IContentModel[]) => {
       try {
         setInProgress({ sectionName, value: true });
-        var instance = activeReport.instances.length ? activeReport.instances[0] : undefined;
-        if (!instance || instance.sentOn) {
-          toast.error('Unable to generate a new report');
-          return;
-        }
+        let currentReport: IReportModel | undefined = { ...report };
+        let instance = currentReport.instances.length ? currentReport.instances[0] : undefined;
 
-        const convertedContent = toInstanceContent(
-          content,
-          activeReport.instances[0].id,
-          sectionName,
-          0,
-        );
+        if (!report.instances.length) {
+          // Fetch the report because the one in memory has no instances.
+          currentReport = await getReport(report.id);
 
-        const instances = activeReport.instances.map((i) =>
-          i.id === instance!.id ? { ...i, content: [...i.content, ...convertedContent] } : i,
-        );
+          // The report does not have an instance, we must create one to add content.
+          // This should only occur the first time after a new report is created.
+          if (!currentReport?.instances.length) {
+            currentReport = await generateReport(report.id, true);
+          }
+        }
 
-        const report = {
-          ...activeReport,
-          instances: instances,
-        };
+        if (
+          instance &&
+          [ReportStatusName.Accepted, ReportStatusName.Completed, ReportStatusName.Failed].includes(
+            instance.status,
+          )
+        ) {
+          // Start the next report.
+          currentReport = await generateReport(report.id, true);
+          instance = currentReport.instances.length ? currentReport.instances[0] : undefined;
+        }
 
-        await addContentToReport(report.id, convertedContent);
+        if (!currentReport || !instance) {
+          toast.error('Failed to generate a new report');
+          return;
+        }
+
+        const convertedContent = toInstanceContent(content, instance.id, sectionName, 0);
+        await addContentToReport(currentReport.id, convertedContent);
 
         toast.success(() => (
           <div>
             {content.length} stories added to report:
-            <Link to={`reports/${report.id}/content`}>{report.name}</Link>
+            <Link to={`reports/${currentReport!.id}/content`}>{currentReport!.name}</Link>
           </div>
         ));
         onClear?.();
-      } catch {}
+      } catch {
+        // Errors are handled globally.
+      } finally {
+        setInProgress({ sectionName: '', value: false });
+      }
     },
-    [activeReport, content, addContentToReport, onClear],
+    [addContentToReport, onClear, getReport, generateReport],
   );
 
-  // ensure no concurrency errors rather than getting from profile store
-  React.useEffect(() => {
```

Comment:
This was very slow and problematic.  We don't need to create new instances when sections are opened.